build.sh

```diff
@@ -0,0 +1,130 @@
+# shell environment; otherwise, they will be given default values, here.
+#
+#  COPR_CONFIG - the path to the configuration file for the COPR account
+#  PODMAN_USR - the username for the container registry account
+#  PODMAN_PSW - the password for the container registry account
+#
+
+PODMAN_USR=${PODMAN_USR-"pbench+mr_jenkins"}
+PODMAN_PSW=${PODMAN_PSW-${DEFAULT_PODMAN_PASSWORD}}
+COPR_CONFIG=${COPR_CONFIG:-"${HOME}/.config/copr"}
+
+# DEBUG
+echo "COPR_CONFIG: $(tr 'A-Za-z' 'N-ZA-Mn-za-m' <<< ${COPR_CONFIG})"
+echo "COPR_CONFIG:copr_url=$(grep copr_url ${COPR_CONFIG})" >&2
+
+# Function to read an initialization/configuration file and print the value of
+# the specified key in the specified section.  Borrowed from
+# https://stackoverflow.com/questions/49399984/parsing-ini-file-in-bash/49400353
+function getIni() {
+  local f=${1}  # Initialization/configuration file name
+  local s=${2}  # Section name
+  local o=${3}  # Option name
+
+  if [[ ! -r ${f} ]]; then
+    echo "Missing or bad COPR config file:  ${f}" >&2
+    exit 1
+  fi
+
+  local rv=$(awk '/^\[.*\]$/{obj=$0}/=/{print obj $0}' ${f} | \
+    sed -En "/${s}]${o}/s/^.*=[[:space:]]*//p" | \
+    tail -n 1)
+
+  if [[ -z ${rv} ]]; then
+    echo "Missing '${o}' in COPR config file, '${f}'" >&2
+    exit 1
+  fi
+
+  echo ${rv}
+}
+
+
+COPR_USER=$(getIni ${COPR_CONFIG} copr-cli username)
+COPR_URL=$(getIni ${COPR_CONFIG} copr-cli copr_url)
+
+if [[ ${COPR_URL} =~ "copr.fedorainfracloud.org" ]]; then
+  URL_PREFIX=https://download.copr.fedorainfracloud.org/results/${COPR_USER}
+elif [[ ${COPR_URL} =~ "copr.devel.redhat.com" ]]; then
+  URL_PREFIX=http://coprbe.devel.redhat.com/results/${COPR_USER}
+else
+  echo "Unexpected COPR_URL: '${COPR_URL}'" >&2
+  exit 1
+fi
+
+# Install the linter requirements and add them to the PATH.
+export PATH=${HOME}/.local/bin:${PATH}
+python3 -m pip install --user -r lint-requirements.txt
+
+# If this script is run in a container and the user in the container doesn't
+# match the owner of the Git checkout, then Git objects; these config settings
+# prevent the complaints.
+GITTOP=$(git rev-parse --show-toplevel 2>&1 | head -n 1)
+if [[ ${GITTOP} =~ "fatal: unsafe repository ('/home/root/pbench'" ]] ; then
+	git config --global --add safe.directory /home/root/pbench
+	git config --global --add safe.directory /home/root/pbench/agent/stockpile
+	GITTOP=$(git rev-parse --show-toplevel)
+fi
+
+# Install the Dashboard dependencies, including the linter's dependencies and
+# the unit test dependencies.
+( cd dashboard && npm install )
+
+# Test for code style and lint
+black --check .
+flake8 .
+( cd dashboard && npx eslint "src/**" --max-warnings 0 )
+
+# Run unit tests
+tox                                     # Agent and Server unit tests and legacy tests
+( cd dashboard && CI=true npm test )    # Dashboard unit tests
+
+# Build Agent and Server RPMs (Server RPM includes the Dashboard) on COPR
+#
+# TODO:  We should set up or maintain a ./seqno file.
```

Comment:
To be honest, I haven't fully contemplated how this thing is going to work on the COPR end.  At this point, concurrent PRs will get mixed up with each other.  And, the builds are going to "build up" on COPR, too, as we have no mechanism for clearing out old ones, presently (although, I think we can make them self-delete with a short expiration time).

The fact that the workspace is not recreated would potentially help out.  That would mean that each subsequent rerun of a particular PR would get a new sequence number, which would help considerably once we start pulling the RPMs.  And, it _might_ mean that the `main` would be properly sequenced without us having to deal with having a `seqno` file in the repo, which would be very cool.

So, yeah, there is definitely a couple of TODO items here.  Fortunately, they don't need to gate this PR.

Reply:
I've added a `seqno` file, and it seems to work.

I looked on COPR, and there is a way to mark a project for auto-deletion, but I don't think we want that:  we want to delete individual builds, not the whole project.  😢  And, I don't see a way to do that...so this may need some attention in the future.